File: csharp/StrangeCharacters/TestCharacters/CharactersTest.cs
using Characters;
using NUnit;
using NUnit.Framework;

namespace TestCharacters;

public class CharactersTest
{
    public List<Character> allTestData()
    {
        var joyce = new Character("Joyce", "Byers");
        var jim = new Character("Jim", "Hopper");
        var mike = new Character("Mike", "Wheeler");
        var eleven = new Character("Eleven");
        var dustin = new Character("Dustin", "Henderson");
        var lucas = new Character("Lucas", "Sinclair");
        var nancy = new Character("Nancy", "Wheeler");
        var jonathan = new Character("Jonathan", "Byers");
        var will = new Character("Will", "Byers");
        var karen = new Character("Karen", "Wheeler");
        var steve = new Character("Steve", "Harrington");
        var mindflayer = new Character("Mindflayer", isMonster:true);
        var demagorgon = new Character("Demagorgon", isMonster:true);
        var demadog = new Character("Demadog", isMonster:true);
        
        joyce.AddChild(jonathan);
        joyce.AddChild(will);
        jim.AddChild(eleven);
        karen.AddChild(nancy);
        karen.AddChild(mike);
        
        eleven.SetNemesis(demagorgon);
        will.SetNemesis(mindflayer);
        dustin.SetNemesis(demadog);

        return new List<Character>()
        {
            joyce,
            jim,
            mike,
            will,
            eleven,
            dustin,
            lucas,
            nancy,
            jonathan,
            mindflayer,
            demagorgon,
            demadog,
            karen,
            steve,
        };
    }

    [TestCase]
    public void FindCharacterByFirstName()
    {
        var finder = new CharacterFinder(allTestData());
        var character = finder.FindByFirstName("Jim");
        Assert.AreEqual("Jim", character?.FirstName);
    }
    
}

File: php/tests/CharactersTest.php
<?php

namespace Emilybache\StrangecharacterTestdesignKata;

use Emilybache\StrangecharacterTestdesignKata\Character;
use Emilybache\StrangecharacterTestdesignKata\CharacterFinder;
use PHPUnit\Framework\TestCase;

class CharactersTest extends TestCase
{
    /**
     * @return Character[]
     */
    public static function allTestData(): array
    {
        $joyce = new Character("Joyce", "Byers");
        $jim = new Character("Jim", "Hopper");
        $mike = new Character("Mike", "Wheeler");
        $eleven = new Character("Eleven");
        $dustin = new Character("Dustin", "Henderson");
        $lucas = new Character("Lucas", "Sinclair");
        $nancy = new Character("Nancy", "Wheeler");
        $jonathan = new Character("Jonathan", "Byers");
        $will = new Character("Will", "Byers");
        $karen = new Character("Karen", "Wheeler");
        $steve = new Character("Steve", "Harrington");
        $mindflayer = Character::newMonster("Mindflayer");
        $demagorgon = Character::newMonster("Demagorgon");
        $demadog = Character::newMonster("Demadog");

        $joyce->addChild($jonathan);
        $joyce->addChild($will);
        $jim->addChild($eleven);
        $karen->addChild($nancy);
        $karen->addChild($mike);

        $eleven->setNemesis($demagorgon);
        $will->setNemesis($mindflayer);
        $dustin->setNemesis($demadog);

        return [
            $joyce,
            $jim,
            $mike,
            $will,
            $eleven,
            $dustin,
            $lucas,
            $nancy,
            $jonathan,
            $mindflayer,
            $demagorgon,
            $demadog,
            $karen,
            $steve
        ];
    }

    public function testFindCharactersByFirstName(): void
    {
        $finder = new CharacterFinder(static::allTestData());
        $character = $finder->findByFirstName("Jim");
        $this->assertNotNull($character);
        $this->assertSame("Jim", $character->firstName);
    }
}
